File: A/alice_and_bob.c
#include <stdio.h>
#include <stdlib.h>

void solve() {
    int n, a;
    scanf("%d %d", &n, &a);
    int v, x, y, l = 0, r = 0;
    for (int i = 0; i < n; ++i) {
        scanf("%d", &v);
        if (v == a) {
            continue;
        }
        x = abs(v - a - 1);
        y = abs(v - a + 1);
        x > y ? ++l : ++r;
    }
    printf("%d\n", (l > r ? --a : ++a));
}

int main(void) {
    int t;
    scanf("%d", &t);
    while (t--) {
        solve();
    }
    return 0;
}

File: A/alice_and_bob.py
import sys
input = sys.stdin.readline

def solve():
    n, a = map(int, input().split())
    arr = list(map(int, input().split()))
    l = r = 0
    for v in arr:
        if v == a:
            continue
        x = abs(v - a - 1)
        y = abs(v - a + 1)
        if x > y:
            l += 1
        else:
            r += 1
    print(a - 1 if l > r else a + 1)

for _ in range(int(input())):
    solve()
